pkg/xds/generator/dns_generator.go

```diff
@@ -53,7 +53,7 @@ func (g DNSGenerator) Generate(ctx context.Context, rs *core_xds.ResourceSet, xd
 		}
 	}
 	if proxy.Metadata.HasFeature(xds_types.FeatureEmbeddedDNS) {
-		dnsInfo := dpapi.DNSProxyConfig{TTL: 3600, Records: []dpapi.DNSRecord{}}
+		dnsInfo := dpapi.DNSProxyConfig{TTL: 30, Records: []dpapi.DNSRecord{}}
```

Comment:
I think I set 3600 because that's like a DNS common default. So I'd add a comment:

```suggestion
                // This is purposefully set to 30s to avoid DNS cache stale with ExternalService and Kong Gateway see: https://github.com/kumahq/kuma/issues/13353.
		dnsInfo := dpapi.DNSProxyConfig{TTL: 30, Records: []dpapi.DNSRecord{}}
```